### PR TITLE
:bug: Use CreateAndWait in MP ScaleFromZero test

### PR DESCRIPTION
**Pull request by killianmuldoon**
Add an eventually to the test `Should delete retired nodes when scaled to zero` for the MachinePool reconciler. This started flaking recently, likely due to changes in the use of the clusterCacheTracker in this reconciler.

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
/area testing

**Comment by killianmuldoon** [comment]
Context: This is the flake https://storage.googleapis.com/k8s-triage/index.html?job=.*-cluster-api-.*&xjob=.*-provider-.*#187cc3ac4b416973fd1f

File: exp/internal/controllers/machinepool_controller_phases_test.go [hunk 1]
```diff
@@ -1249,13 +1249,14 @@ func TestReconcileMachinePoolScaleToFromZero(t *testing.T) {
...
 		r.reconcilePhase(machinepool)
-
 		g.Expect(machinepool.Status.GetTypedPhase()).To(Equal(expv1.MachinePoolPhaseRunning))
 
 		delNode := &corev1.Node{}
-		err = env.GetAPIReader().Get(ctx, client.ObjectKeyFromObject(node), delNode)
-		g.Expect(err).ToNot(BeNil())
-		g.Expect(apierrors.IsNotFound(err)).To(BeTrue())
+		g.Eventually(func(g Gomega) {
+			err = fakeClient.Get(ctx, client.ObjectKeyFromObject(node), delNode)
```
Comment (sbueringer) [inline exp/internal/controllers/machinepool_controller_phases_test.go:1256]: as far as I can tell the node is never created in the fakeClient. I think we can't switch from an env client to the fake client here

Comment (killianmuldoon) [reply exp/internal/controllers/machinepool_controller_phases_test.go:1256]: Completely right - let me fix that.

/hold

**Comment by killianmuldoon** [comment]
/retitle :bug: Use CreateAndWait in MP ScaleFromZero test

**Comment by killianmuldoon** [comment]
/hold cancel

Test green for me with ~30K runs on my machine. I caught the flake in action and it seems like it was down to the node not being available when reconcile was called.

**Review by fabriziopandini — COMMENTED** [review]
Nice fix, again kudos for the relentless work on flakes
/lgtm
/approve
